server/graphql/v2/mutation/PaymentMethodMutations.js

```diff
@@ -1,39 +1,107 @@
-import { GraphQLBoolean, GraphQLNonNull, GraphQLString } from 'graphql';
+import { GraphQLBoolean, GraphQLNonNull, GraphQLObjectType, GraphQLString } from 'graphql';
 import { pick } from 'lodash';
 
 import models from '../../../models';
 import { setupCreditCard } from '../../../paymentProviders/stripe/creditcard';
-import { Forbidden, ValidationFailed } from '../../errors';
+import { Forbidden } from '../../errors';
 import { AccountReferenceInput, fetchAccountWithReference } from '../input/AccountReferenceInput';
 import { CreditCardCreateInput } from '../input/CreditCardCreateInput';
 import { PaymentMethodCreateInput } from '../input/PaymentMethodCreateInput';
+import { fetchPaymentMethodWithReference, PaymentMethodReferenceInput } from '../input/PaymentMethodReferenceInput';
 import { PaymentMethod } from '../object/PaymentMethod';
+import { StripeError } from '../object/StripeError';
 
-const addCreditCard = {
-  type: new GraphQLNonNull(PaymentMethod),
+const CreditCardWithStripeError = new GraphQLObjectType({
+  name: 'CreditCardWithStripeError',
+  fields: {
+    paymentMethod: {
+      type: new GraphQLNonNull(PaymentMethod),
+      description: 'The payment method created',
+    },
+    stripeError: {
+      type: StripeError,
+      description: 'This field will be set if there was an error with Stripe during strong customer authentication',
+    },
+    shouldBeSaved: {
+      type: GraphQLBoolean,
+      description: 'If the Payment Method needs to be marked as saved again during confirmCreditCard',
+    },
+  },
+});
+
+const addStripeCreditCard = {
+  type: PaymentMethod,
   description: 'Add a new payment method to be used with an Order',
   args: {
     paymentMethod: {
       type: PaymentMethodCreateInput,
       description: 'A Payment Method to add to an Account',
-      deprecationReason: '2020-08-24: Please use creditCardInfo',
     },
+    account: {
+      type: AccountReferenceInput,
+      description: 'Account to add Payment Method to',
+    },
+  },
+  async resolve(_, args) {
+    const account = await fetchAccountWithReference(args.account);
+    const collective = await models.Collective.findByPk(account.id);
+    if (!collective) {
+      throw Error('This collective does not exist');
```

Comment:
We should say "account"? Not collective.

Reply:
Sorry, re: `addStripeCreditCard`, we're only adding back a deprecated mutation briefly to make sure there are no breaking changes. Then in a follow up API PR we will remove it (once the frontend is deployed using `addCreditCard`) immediately. I can make the changes to the code but we shouldn't dwell on 'perfecting' it.

Reply:
Ok